# <b> Results from 2023 Financial Reports (Q1, Q2, Q3, Q4)</b>

In [40]:
#testing QTR1 first

import pandas as pd
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/output/Results_QTR1.csv")
df.describe()

filename  \
count                                                2521   
unique                                                734   
top     20230331_10-K_edgar_data_1280263_0000950170-23...   
freq                                                   52   

                                                   Item1A  \
count                                                2521   
unique                                                726   
top     of part i risk factors, item of part ii manage...   
freq                                                   52   

                                                 Sentence  \
count                                                2521   
unique                                               2435   
top     additionally, several states and localities ha...   
freq                                                    5   

                                              SentencesR3  \
count                                                2521   
unique                                               2500   
top     a number of recent technologies have worked wi...   
freq                                                    3   

                                              SentencesR5  \
count                                                2521   
unique                                               2504   
top     these disruptions or the company's failure to ...   
freq                                                    2   

               KeywordMentioned  
count                      2521  
unique                       93  
top     artificial intelligence  
freq                        655

So about 734 files are ones which have AI related keywords in them.

In [41]:
# check how many files orginally present in the concerned folder,i.e QTR1 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR1'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR1: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR1: 7145
Number of relevant files extracted: 734
Percentage of relevant files: 10.27%


So out of 7k+ financial reports, about 730 are those whose risk factors are related to AI.

In [7]:
df.head()

filename  \
0  20230301_10-K_edgar_data_912752_0000912752-23-...   
1  20230301_10-K_edgar_data_1831915_0000950170-23...   
2  20230223_10-K_edgar_data_1666134_0001666134-23...   
3  20230221_10-K_edgar_data_1669811_0000950170-23...   
4  20230221_10-K_edgar_data_1669811_0000950170-23...   

                                              Item1A  \
0  . risk factors , item . management s discussio...   
1  . risk factors our operations and financial re...   
2  . risk factors and elsewhere herein. forward-l...   
3  . risk factors of this annual report, in addit...   
4  . risk factors of this annual report, in addit...   

                                            Sentence  \
0  we have also partnered with several other comp...   
1  furthermore, in europe, there is a proposed re...   
2  embedded machine learning then reduces the man...   
3  via integration with the company s ebrevia sol...   
4  in , the company sold its language solutions b...   

                                         SentencesR3  \
0  other assets within the consolidated financial...   
1  for example, the european union s general data...   
2  blackline cash application transforms the orde...   
3  the company s venue solution is a highly secur...   
4  company history on october , , dfin became an ...   

                                         SentencesR5         KeywordMentioned  
0  see note . nature of operations and summary of...  artificial intelligence  
1  if we become subject to new data privacy and s...  artificial intelligence  
2  this critical process is often highly manual. ...         machine learning  
3  for m a transactions, the company supports dea...  artificial intelligence  
4  the company s tech-enabled services offerings ...  artificial intelligence

Randomly Checking the content extracted - sentence & the AI term used: 

In [19]:
content = df.loc[1,'Item1A']
content

'. risk factors our operations and financial results are subject to numerous risks and uncertainties, including those described below, which may have a material and adverse effect on our business, results of operations, cash flows, financial conditions, and the trading price of our common stock. the risks and uncertainties described below are not the only ones facing us. additional risks and uncertainties not presently known to us or that we currently deem immaterial also may impair our business operations. you should consider these risks and uncertainties carefully, together with all of the other information included or incorporated by reference in this annual report on form -k. if any of the following risks actually occur, our business, financial condition, results of operations and future prospects could be materially and adversely affected. you should not interpret our disclosure of any of the following risks to imply that such risks have not already materialized. summary risk fact

In [26]:
Sentence = df.loc[1,'Sentence']
print(Sentence)

Keyword = df.loc[1,'KeywordMentioned']
print(Keyword)

furthermore, in europe, there is a proposed regulation related to artificial intelligence ai that, if adopted, could impose onerous obligations related to the use of ai-related systems.
artificial intelligence


Hence, here AI Sentence extracted from risk factors is: 
"...there is a proposed regulation related to artificial intelligence ai that..." and keyword is "artificial intelligence"

In [28]:
content = df.loc[2,'Item1A']
content

'. risk factors and elsewhere herein. forward-looking statements should not be read as a guarantee of future performance or results, and you should not place undue reliance on such statements. furthermore, we undertake no obligation to revise or update any forward-looking statements for any reason, except as required by applicable law. unless the context otherwise requires, the terms blackline, inc., the company, we, us and our in this annual report on form -k refer to the consolidated operations of blackline, inc. and its consolidated subsidiaries as a whole. item . business overview we have created comprehensive cloud-based solutions designed to transform and modernize accounting and finance operations for mid-market and enterprise organizations in all industries globally. our secure, scalable solutions support critical financial close, accounts receivable and intercompany accounting processes. by introducing software to automate these processes and to enable them to function continu

In [29]:
Sentence = df.loc[2,'Sentence']
print(Sentence)

Keyword = df.loc[2,'KeywordMentioned']
print(Keyword)

embedded machine learning then reduces the manual effort involved in the process and releases working capital for our customers.
machine learning


Hence, AI sentence extracted from risk factor is:
"...embedded machine learning then reduces the manual effort involved in the process and releases working capital..." & keyword is "machine learning"

Let's see what unique keywords have been extracted to double-verify if they are AI related:

In [30]:
unique_keywords = df['KeywordMentioned'].unique()
for i in unique_keywords:
    print(i)

artificial intelligence
machine learning
generative ai, artificial intelligence, machine learning
generative ai
generative ai, artificial intelligence
business intelligence
artificial intelligence, machine learning
data scientist
aiml
data science
data science, business intelligence
generative ai, machine learning
artificial intelligence, machine learning, data scientist
machine learning, data science
natural language processing, chatbot
chatbot
natural language processing
computer vision
data mining
machine learning, computer vision
big data
financial forecasting
artificial intelligence, machine learning, computer vision
algorithmic trading
machine learning, image recognition
machine learning, business intelligence
ai/ml, artificial intelligence, machine learning
ai/ml
deep learning
regression analysis
machine learning, big data
artificial intelligence, machine learning, data mining
a.i., artificial intelligence
a.i.
hadoop
machine learning, hadoop
deep learning, data science
artifici

## QTR2:

In [42]:
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/output/Results_QTR2.csv")
df.describe()

filename  \
count                                                 351   
unique                                                 87   
top     20230622_10-K_edgar_data_1577526_0001628280-23...   
freq                                                   39   

                                                   Item1A  \
count                                                 351   
unique                                                 87   
top     of this annual report on form -k and those els...   
freq                                                   39   

                                                 Sentence  \
count                                                 351   
unique                                                347   
top     the general compute and artificial intelligenc...   
freq                                                    2   

                                              SentencesR3  \
count                                                 351   
unique                                                351   
top     as a result, we relinquished to the fcc our na...   
freq                                                    1   

                                              SentencesR5  \
count                                                 351   
unique                                                351   
top     as a result, the fcc is currently holding over...   
freq                                                    1   

               KeywordMentioned  
count                       351  
unique                       43  
top     artificial intelligence  
freq                         63

In [32]:
df.head()

filename  \
0  20230417_10-K_edgar_data_1850051_0001104659-23...   
1  20230417_10-K_edgar_data_1850051_0001104659-23...   
2  20230417_10-K_edgar_data_1850051_0001104659-23...   
3  20230417_10-K_edgar_data_1850051_0001104659-23...   
4  20230417_10-K_edgar_data_1850051_0001104659-23...   

                                              Item1A  \
0  . risk factors. an investment in our securitie...   
1  . risk factors. an investment in our securitie...   
2  . risk factors. an investment in our securitie...   
3  . risk factors. an investment in our securitie...   
4  . risk factors. an investment in our securitie...   

                                            Sentence  \
0  although we intend to focus on identifying com...   
1  we will seek to complete an initial business c...   
2  if we pursue a target business in the artifici...   
3  business combinations with companies at the in...   
4  however, our efforts in identifying prospectiv...   

                                         SentencesR3  \
0  attractive deals could also become scarcer for...   
1  in the event of an exercise on a cashless basi...   
2  if there is an annual meeting, as a consequenc...   
3  in addition, we have agreed not to enter into ...   
4  business combinations with companies at the in...   

                                         SentencesR5         KeywordMentioned  
0  as a result, at times, fewer attractive target...  artificial intelligence  
1  second, if a registration statement covering t...  artificial intelligence  
2  in addition, our board of directors, whose mem...  artificial intelligence  
3  we may not hold an annual meeting of stockhold...  artificial intelligence  
4  accordingly, our initial stockholders will con...  artificial intelligence

In [43]:
# check how many files orginally present in the concerned folder,i.e QTR2 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR2'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR2: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR2: 7564
Number of relevant files extracted: 87
Percentage of relevant files: 1.15%


In [44]:
unique_keywords = df['KeywordMentioned'].unique()
for i in unique_keywords:
    print(i)

artificial intelligence
machine learning, big data
artificial intelligence, machine learning
data mining
machine learning
data science
machine learning, data science
financial forecasting
natural language processing
deep learning
big data
business intelligence
ai/ml
data science, data scientist
aiml
recommender system, data science
computer vision, deep learning
recommender system
artificial intelligence, chatbot
artificial intelligence, machine learning, natural language processing
artificial intelligence, machine learning, data mining
computer vision
ai/ml, artificial intelligence, machine learning
ai/ml, artificial intelligence
large language model
ai/ml, machine learning
generative ai
data scientist
artificial intelligence, machine learning, data science
artificial intelligence, computer vision, deep learning
generative ai, ai/ml, artificial intelligence, machine learning
generative ai, ai/ml
generative ai, machine learning
 llm
supervised learning, unsupervised learning, deep lear

## QTR3:

In [45]:
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/output/Results_QTR3.csv")
df.describe()

filename  \
count                                                 346   
unique                                                 71   
top     20230901_10-K_edgar_data_1327567_0001327567-23...   
freq                                                   19   

                                                   Item1A  \
count                                                 346   
unique                                                 71   
top     and elsewhere in this annual report on form -k...   
freq                                                   19   

                                                 Sentence  \
count                                                 346   
unique                                                344   
top     some of our competitors may also expand their ...   
freq                                                    2   

                                              SentencesR3  \
count                                                 346   
unique                                                346   
top     risks related to our business operations and i...   
freq                                                    1   

                                              SentencesR5  \
count                                                 346   
unique                                                346   
top     costs could also increase as a result of healt...   
freq                                                    1   

               KeywordMentioned  
count                       346  
unique                       36  
top     artificial intelligence  
freq                        103

In [46]:
df.head()

filename  \
0  20230811_10-K_edgar_data_943819_0000943819-23-...   
1  20230811_10-K_edgar_data_943819_0000943819-23-...   
2  20230721_10-K_edgar_data_866729_0000866729-23-...   
3  20230815_10-K_edgar_data_1049521_0001049521-23...   
4  20230725_10-K_edgar_data_1679273_0001558370-23...   

                                              Item1A  \
0  risk factors and elsewhere in this report. inf...   
1  risk factors and elsewhere in this report. inf...   
2  | risk factors set forth below and elsewhere i...   
3  . risk factors: risks related to business oper...   
4  . risk factors in this form -k. we caution rea...   

                                            Sentence  \
0  the ftc also has focused on the use of online ...   
1  certain of our products and services include t...   
2  there are substantial uncertainties associated...   
3  customers conditioned to seek quarter-end disc...   
4  at avantor, ms. rajagopalan led the company s ...   

                                         SentencesR3  \
0  the federal trade commission, or ftc, and stat...   
1  failure to comply with these regulations could...   
2  if saas providers are unable to provide these ...   
3  high quarterly book-ship ratios pressure our i...   
4  she also has held various sales positions with...   

                                         SentencesR5         KeywordMentioned  
0  in addition, congress and some states are cons...  artificial intelligence  
1  our business activities are subject to extensi...  artificial intelligence  
2  the company outsources business processes to r...  artificial intelligence  
3  any significant change in our customers purcha...    financial forecasting  
4  since joining conagra in , ms. miller has held...  artificial intelligence

In [47]:
# check how many files orginally present in the concerned folder,i.e QTR3 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR3'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR3: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR3: 6583
Number of relevant files extracted: 71
Percentage of relevant files: 1.08%


In [48]:
unique_keywords = df['KeywordMentioned'].unique()
for i in unique_keywords:
    print(i)

artificial intelligence
financial forecasting
artificial intelligence, machine learning
machine learning
business intelligence
computer vision, natural language processing
machine learning, natural language processing
generative ai
generative ai, machine learning
large language model, machine learning, natural language processing
artificial intelligence, data science
chatbot
ai/ml, computer vision
artificial intelligence, machine learning, natural language processing
generative ai, artificial intelligence
data science
ai/ml
deep learning
artificial intelligence, machine learning, big data
generative ai, artificial intelligence, machine learning
big data
generative ai, ai/ml
ai/ml, artificial intelligence, machine learning
data scientist
large language model
generative ai, machine learning, deep learning
generative ai, large language model, machine learning
computer vision
machine learning, deep learning
machine learning, computer vision
artificial intelligence, machine learning, comput

## QTR4:

In [49]:
df = pd.read_csv("/Users/rravindra0463/Desktop/ai_proj/venv/output/Results_QTR4.csv")
df.describe()

filename  \
count                                                 294   
unique                                                 77   
top     20231211_10-K_edgar_data_1837240_0001837240-23...   
freq                                                   20   

                                                   Item1A  \
count                                                 294   
unique                                                 77   
top     risk factors and part ii, item management s di...   
freq                                                   20   

                                                 Sentence  \
count                                                 294   
unique                                                278   
top     new laws that may impact our business, such as...   
freq                                                    2   

                                              SentencesR3  \
count                                                 294   
unique                                                294   
top     other regulators have expressed interest in th...   
freq                                                    1   

                                              SentencesR5  \
count                                                 294   
unique                                                294   
top     other existing laws such as the ccpa, future a...   
freq                                                    1   

               KeywordMentioned  
count                       294  
unique                       38  
top     artificial intelligence  
freq                        103

In [50]:
# check how many files orginally present in the concerned folder,i.e QTR4 here
import os
folder_path = '/Users/rravindra0463/Desktop/ai_proj/venv/QTR4'

# list all files in the folder
files = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f))]

# count the number of files
num_files = len(files)
print(f'Number of files in the folder QTR4: {num_files}')

# calculate metrics
unique_filenames = df['filename'].nunique()
percentage = (unique_filenames/ num_files) * 100

print(f'Number of relevant files extracted: {unique_filenames}')
print(f'Percentage of relevant files: {percentage:.2f}%')

Number of files in the folder QTR4: 6565
Number of relevant files extracted: 77
Percentage of relevant files: 1.17%


In [51]:
unique_keywords = df['KeywordMentioned'].unique()
for i in unique_keywords:
    print(i)

artificial intelligence
artificial intelligence, data science
machine learning
artificial intelligence, machine learning
machine learning, big data
financial forecasting
generative ai
business intelligence
ai/ml
data science
a.i.
a.i., machine learning
big data
aiml
generative ai, machine learning
data scientist
artificial intelligence, machine learning, big data
large language model,  llm
 llm
 llm, artificial intelligence
algorithmic trading
artificial intelligence, business intelligence
generative ai, chatbot
deep learning
generative ai, large language model
large language model, artificial intelligence
natural language processing
generative ai,  llm
computer vision
ai/ml, artificial intelligence, machine learning
ai/ml, artificial intelligence
machine learning, data mining
artificial intelligence, natural language processing, chatbot
generative ai, artificial intelligence
machine learning, data science
ai/ml, machine learning
large language model
chatbot


### ------------------------------------------------------------------------ ###

<a href="https://drive.google.com/drive/folders/1-wpBG2rm4uG1lHidT3GddxIcE2awBAqG?usp=drive_link" target="_blank">Access all the csv files here!</a>
